src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -2,96 +2,143 @@
+                      ALIGNMENT_VECTOR_LENGTH_FACTOR * alignment_vector;
 
         align_placement_tactic->updateControlParams(
             setup_point, setup_angle, TbotsProto::DribblerMode::OFF,
             TbotsProto::BallCollisionType::AVOID, {AutoChipOrKickMode::OFF, 0},
-            TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+            TbotsProto::MaxAllowedSpeedMode::STOP_COMMAND,
             TbotsProto::ObstacleAvoidanceMode::SAFE);
 
-        tactics_to_run[0].emplace_back(align_placement_tactic);
-
-        event.common.set_tactics(tactics_to_run);
+        tactics_to_run[0].push_back(align_placement_tactic);
     }
+
+    setupMoveTactics(event, event.common.num_tactics -
+                                static_cast<unsigned int>(tactics_to_run[0].size()));
+    tactics_to_run[0].insert(tactics_to_run[0].end(), move_tactics.begin(),
+                             move_tactics.end());
+
+    event.common.set_tactics(tactics_to_run);
 }
 
 void BallPlacementPlayFSM::placeBall(const Update &event)
 {
     PriorityTacticVector tactics_to_run = {{}};
 
-    // setup move tactics for robots away from ball placing robot
-    setupMoveTactics(event);
-    tactics_to_run[0].insert(tactics_to_run[0].end(), move_tactics.begin(),
-                             move_tactics.end());
-
-    Angle final_angle = Angle::zero();
     std::optional<Point> placement_point =
         event.common.world_ptr->gameState().getBallPlacementPoint();
 
-    Vector placement_dribble_vector;
     if (placement_point.has_value())
     {
-        placement_dribble_vector =
-            placement_point.value() - event.common.world_ptr->ball().position();
-        final_angle = placement_dribble_vector.orientation();
+        Point ball_pos = event.common.world_ptr->ball().position();
+        std::optional<Robot> robot_placing_ball =
+            event.common.world_ptr->friendlyTeam().getNearestRobot(ball_pos);
+
+        if (robot_placing_ball.has_value())
+        {
+            Angle final_angle = robot_placing_ball->orientation();
+
+            Vector placement_dribble_vector =
+                placement_point.value() - event.common.world_ptr->ball().position();
+            if (placement_dribble_vector.length() > 0.3)
+            {
+                final_angle = placement_dribble_vector.orientation();
+            }
+
+            place_ball_tactic->updateControlParams(
+                event.common.world_ptr->gameState().getBallPlacementPoint(), final_angle,
+                true, TbotsProto::MaxAllowedSpeedMode::DRIBBLE);
+
+            tactics_to_run[0].push_back(place_ball_tactic);
+        }
     }
 
-    // setup ball placement tactic for ball placing robot
-    place_ball_tactic->updateControlParams(
-        event.common.world_ptr->gameState().getBallPlacementPoint(), final_angle, true);
-    tactics_to_run[0].emplace_back(place_ball_tactic);
+    setupMoveTactics(event, event.common.num_tactics -
```

Comment:
needs safety checks that event.common.num_tactics > tactics_to_run[0].size(). Otherwise, we risk underflowing into an unsigned type.